src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -11,7 +12,8 @@ namespace Microsoft.PowerFx.Types
 {
     /// <summary>
     /// Represent a Record. Records have named fields which can be other values. 
-    /// </summary>
+    /// </summary>    
+    [SuppressMessage("Design", "CA1012:Abstract type should not have public constructors", Justification = "n/a")]
```

Comment:
>[SuppressMessage("Design", "CA1012:Abstract type should not have public constructors", Justification = "n/a")] [](http://example.com/codeflow?start=4&length=110)

We can make the .ctor protected and remove this suppression

Reply:
Doesn't work well with static Empty() and other methods